pkg/manager/member/pd_member_manager.go

```diff
@@ -463,6 +463,15 @@ func (pmm *pdMemberManager) getNewPDServiceForTidbCluster(tc *v1alpha1.TidbClust
 			Selector: pdLabel,
 		},
 	}
+	// if set pd service type ,overwrite global variable services
+	svcSpec := tc.Spec.PD.Service
+	if svcSpec != nil {
+		pdService.Spec.ClusterIP = svcSpec.ClusterIP
+		pdService.Spec.Type = svcSpec.Type
+		pdService.Spec.LoadBalancerIP = svcSpec.LoadBalancerIP
+		pdService.ObjectMeta.Annotations = svcSpec.Annotations
```

Comment:
what if the fields in svcSpec are empty strings? We should check it first

Reply:
you re right, the type need to check, other values I think empty is not affect.